src/examples/heading/levels.man
.TH "HEADING 1" "" "April 2017" "" ""
.SH "NAME"
\fBheading 1\fR
.SH heading 2
.SS heading 3
.SS heading 4
.SS heading 5
.SS heading 6

Changelog.md
Version changes
=================================================

The following list gives a short overview about what is changed between
individual versions:

Version 3.0.0 (2017-04-07)
-------------------------------------------------
New major realease with completely changed internal structure and some special features missing.

- Remove clean-css.
- Update some packages.
- Cache loaded and setup styles.
- Remove examples from npm package.
- Optimized and fixed style inclusion to include every one once.
- Fix segmentation fault in report by explicitly calling the callback.
- Remove unneccessary line.
- Fix inclusion of local html styles.
- Optimize formatter call.
- Updated inline docs.
- Add automatic read from file.
- Disable fuzzy linkify.
- Fix output format access.
- Fix lint errors.
- Fix format error.
- Update search for api methods.
- Add navigation methods to api.
- Update  async@2.1.5, debug@2.6.3, twemoji@2.2.5, highlight.js@9.10.0, markdown-it@8.3.1, moment@2.18.1, clean-css@4.0.9, mermaid@7.0.0, memoizee@0.4.4
- Allow title setting throughz style.
- Add support for toc.
- Completed typograph test and examples.
- Add typographic replacements and language specification using styles.
- Add support for mark, insert, sub and sup formatting.
- Added auto create element for include.
- Simple includes working.
- Finished boxes implementation.
- Added api for boxes.
- Finish html layout for simple boxes.
- Allow multibox in console and text output.
- Add heading anchor check.
- Smaller fixes to make all tests work again.
- Add formatter for text boxes.
- Begin text and html formatter.
- Implement markdown parsing and formatting of boxes.
- Start new plugin for box.
- Support kramdown style syntax.
- Add api for definition list.
- Add container markdown parsing.
- Add output formats for definition list.
- Implement markdown for definition lists.
- Fix target depth in html style.
- Add styles for html decoration.
- Fixed html markdown test to support new comment elements.
- Add more list api tests.
- Completed html, md, text, console outputs for basic elements.
- Fixed output of task lists.
- Make test for html list layout.
- Add more examples.
- Optimize fixed html output.
- Add more highlighting examples.
- Add highlighting to console output.
- Test pdf export.
- Add more examples of html code highlighting.
- Small fixes.
- Add syntax highlighting for code blocks.
- More examples blockquote.
- Add html2png conversion.
- Add stylesheets to html output.
- Add plugin for gfm tasklists.
- Finish table API with tests.
- Implement api for alinex-table.
- Finish html, text and console output for table.
- Add more colors to console output.
- Implement ansi escapes.
- Allow escaped pipe in table.
- Add strikethrough implementation.
- Api for raw, link and image.
- Support keep breaks (softbreak) setting in html output.
- Allow switch to use references on markdown output.
- All commonmark tests added.
- Add tests for linebreaks.
- More tests for autolink and inline html.
- More tests.
- Complete all element tests.
- Add more text escape tests.
- Merge branch 'v3' of https://github.com/alinex/node-report into v3
- Add image markdown tests.
- Complete link tests.
- Complete tests for html blocks.
- Added html tag transforming from/to markdown.
- Complete link markdown tests.
- Fix formatting of image and link also if nested.
- Some more link tests ans initial image parser implementation.
- Implement links in markdown.
- Finished markdown emphasis tests.
- More emphasis tests.
- Some emphasis/strong tests.
- Optimize API and add fixed().
- Optimize masking of characters.
- Finish list tests.
- Implement loose and tight lists with more tokenlist elements.
- Support tight list parsing and writting.
- A lot more list tests.
- Start list tests.
- Add tests for text (tabs).
- Update all blockquote tests.
- Add tests for paragraph.
- Fix set after close of token.
- Small fixes in man conversion of code and preformatted.
- Added markdown parsing tests for fixed.
- Run all current tests.
- Only rtrim newlines in preformatted text.
- Complete thematic break tests.
- Start tests for thematic break.
- Work with complete heading api.
- Set marker on api close of heading.
- Heading with API in one step.
- First heading API tests with new structure.
- Support more heading specifica.
- Add support for preformatted, softbreak and hardbreak.
- Fix duplicate masking of # to prevent heading.
- First tests to pass reparse check with markdown.
- Integrate parsing through markdown-it into own tokenlist.
- redesign.
- Next redesign phase started.
- Some more tests.
- Small fix.
- Made all current tests pass.
- Redesign with some running tests.
- Redesign emphasis rules.
- Allow inner marker in underscore.
- Some more tests.
- First tests for strong.
- Completed fixed style tests.
- First tests for fixed style.
- Finished code tests.
- More tests for code blocks.
- Closing code block at end of document automatically.
- Enable language definition in code tags and alias through config.
- Change code parser.
- New element code added.
- Simplified list RegExp.
- Added more tests for lists.
- Add more list tests.
- Add some fixes for lists.
- Add more tests for lists.
- Run all tests for blockquotes.
- Fix some more blockquote problems with lazyness.
- Add some blockquote tests.
- More debug output in formatter.
- Optimize token list display in test.
- Complete tests for preformatted.
- Blank lines won't break current predefined section.
- Add moment.js package.
- Add more roff formatters.
- Add more tests for preformatted text.
- Run first tests for preformatted text.
- Trim text output.
- Fix output for formats without page header and footer.
- update todo list
- Update documentation.
- Complete thematic break tests.
- Run all commonmark examples for headings.
- Allow relative paths in include.
- Basic implementation of includes.
- Add anchor to headings for direct call within links.

src/examples/heading/levels.tokens.js
TokenList {
  data: 
   [ { type: 'document', nesting: 1, level: 0 },
     { type: 'heading',
       nesting: 1,
       heading: 1,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 1',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 1,
       level: 1,
       parent: [Object] },
     { type: 'heading',
       nesting: 1,
       heading: 2,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 2',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 2,
       level: 1,
       parent: [Object] },
     { type: 'heading',
       nesting: 1,
       heading: 3,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 3',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 3,
       level: 1,
       parent: [Object] },
     { type: 'heading',
       nesting: 1,
       heading: 4,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 4',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 4,
       level: 1,
       parent: [Object] },
     { type: 'heading',
       nesting: 1,
       heading: 5,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 5',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 5,
       level: 1,
       parent: [Object] },
     { type: 'heading',
       nesting: 1,
       heading: 6,
       level: 1,
       parent: [Object] },
     { type: 'text',
       nesting: 0,
       content: 'heading 6',
       level: 2,
       parent: [Object] },
     { type: 'heading',
       nesting: -1,
       heading: 6,
       level: 1,
       parent: [Object] },
     { type: 'document', nesting: -1, level: 0 } ],
  pos: 19,
  token: 
   { type: 'heading',
     nesting: -1,
     heading: 6,
     level: 1,
     parent: { type: 'document', nesting: 1, level: 0 } } }